### PR TITLE
ADR 31

### PR DESCRIPTION
**Pull request by ErikOstlund**
ADR for RotatingRectangle Tool and TemporalRotatingRectangle Tool

Package: `lib-classifier`

Describe your changes:
Created a ADR to document how two new rectangle drawing tools will be created in the `front-end-monorepo`.
 - RotatingRectangle
 - TemporalRotatingRectangle

### CODE REVIEW

File: docs/arch/adr-31.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+\*everything the `Rectangle` tool has plus this addition:
+
+```json
+{
+  "angle": number
+}
+```
+
+**NOTE:** angle is Degrees (-180, 180)
```
Comment (CKrawczyk) [inline docs/arch/adr-31.md:40]: This should also note that positive angles rotate from `+x` to `+y` (a right-handed coordinate system).  If using SVG with the origin in the upper left, and `+y` downward positive angles rotate clockwise.  See a full discussion of this in https://github.com/zooniverse/front-end-monorepo/issues/500 

The rotation should also be about the center of the rectangle.

File: docs/arch/adr-31.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Status
+
+In Review
+
+## Context
+
+The Solar Jets project requires a TemporalRotatingRectangle drawing tool for volunteers to capture the width, height, angle and time properties of a solar jet. At the time of this writing, the `front-end-monorepo` has a standard Rectangle drawing tool which gives the width and height properties. Based on the standard Rectangle drawing tool, we want to create a RotatingRectangle drawing tool and a TemporalRotatingRectangle drawing tool.
+
+The RotatingRectangle drawing tool is a common tool used in the Panoptes code based and eventually needs to be written in the `front-end-monorepo`. This is a good opportunity to complete this task.
```
Comment (goplayoutside3) [inline docs/arch/adr-31.md:11]:
```suggestion
The RotatingRectangle drawing tool is a common tool used in the Panoptes code base and eventually needs to be written in the `front-end-monorepo`. This is a good opportunity to complete this task.
```

File: docs/arch/adr-31.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Lastly, we will extend the `RotatingRectangle` to create a new `TemporalRotatingRectangle` adding the `displayTime` property to the model.
+
+An example of this tool extension can be seen in this PR: [PR#2099](https://github.com/zooniverse/front-end-monorepo/pull/2099)
+
+## Consequences
+
+Theses decisions allow for these improvements:
+
+- Separation of concerns. Each tool will be it's own component making debugging and testing easier.
```
Comment (goplayoutside3) [inline docs/arch/adr-31.md:84]:
```suggestion
- Separation of concerns. Each tool will be its own component making debugging and testing easier.
```

File: docs/arch/adr-31.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+In Review
+
+## Context
+
+The Solar Jets project requires a TemporalRotatingRectangle drawing tool for volunteers to capture the width, height, angle and time properties of a solar jet. At the time of this writing, the `front-end-monorepo` has a standard Rectangle drawing tool which gives the width and height properties. Based on the standard Rectangle drawing tool, we want to create a RotatingRectangle drawing tool and a TemporalRotatingRectangle drawing tool.
+
+The RotatingRectangle drawing tool is a common tool used in the Panoptes code based and eventually needs to be written in the `front-end-monorepo`. This is a good opportunity to complete this task.
+
+The idea is to create two new rectangle drawing tools based off the standard Rectangle drawing tool. once complete, the `front-end-monorepo` will have the following rectangle drawing tools:
```
Comment (goplayoutside3) [inline docs/arch/adr-31.md:13]:
```suggestion
The idea is to create two new rectangle drawing tools based off the standard Rectangle drawing tool. Once complete, the `front-end-monorepo` will have the following rectangle drawing tools:
```

**Review by goplayoutside3 — APPROVED** [review]
Looks good! I added just a couple typo suggestions for clarity.

File: docs/arch/adr-31.md (hunk 2 above)
Comment (srallen) [reply docs/arch/adr-31.md:11]: Change `Panoptes` to `Panoptes-Front-End` for clarity here too.

File: docs/arch/adr-31.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+```json
+{
+  "displayTime": string
+}
+```
+
+**NOTE:** displayTime will be formatted as:
+
+- If minutes exist: mm:ss:ms
```
Comment (srallen) [inline docs/arch/adr-31.md:57]: Is the implication here that there could be video subjects that are minutes long that projects want volunteers to classify on? I hesitate to support this. We found in past projects that volunteers lost interest on 30 second videos plus the longer the video, the larger the file size will be per subject. Chimp & See found the sweet spot to be around 10 seconds. What are the requirements for Solar Jets? If we can limit this to `ss:ms`, then I think we should.

File: docs/arch/adr-31.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+Theses decisions allow for these improvements:
+
+- Separation of concerns. Each tool will be it's own component making debugging and testing easier.
+- For down-stream aggregation this method is clean as rotating shapes are handled differently than non-rotating ones (the difference stems from how the "distance" between two angles is different than the "distance" between two points because angles wrap back around at 360 deg).
+  Also, the auto-configuration step for aggregation only works with the workflow data, and that is the step that needs to pick between the non-rotating extractor and the rotating extractor (as it is currently written), so having them be different tool types would require less reworking of how aggregation works with the data.
+
+Potential negative impact:
+
+- May cause configuration issues in a new project builder design
```
Comment (srallen) [inline docs/arch/adr-31.md:90]: Could this be elaborated? A few thoughts I have:

Do we support both standard drawing tools and temporary drawing tools for multi-frame subjects that have both video and images? Would this introduce complexity to the project builder UI, classification UI, as well as downstream aggregation? 

How do we distinguish visually between the temporal tools and standard tools? Should we have a separate icon set or another indicator of some sort in the project builder and classifier task area that a temporal version of the tool is loaded?

Will we eventually support temporal versions of all our drawing tools?

I don't expect you to have the answer to these right now, but I think we should note these here as potential consequences and open questions.

**Review by srallen — COMMENTED** [review]
I have a few questions / comments that I've noted inline. Thanks!

File: docs/arch/adr-31.md (hunk 5 above)
Comment (ErikOstlund) [reply docs/arch/adr-31.md:57]: I am trying to cover a wider range of video lengths while keeping the video controller UI clean for shorter videos. This is already built in and lives in the `FormattedTime.js` file.

File: docs/arch/adr-31.md (hunk 6 above)
Comment (ErikOstlund) [reply docs/arch/adr-31.md:90]: Good questions. You're correct; I don't have answers to these. But I will add them to the ADR so we can refer back to them.

File: docs/arch/adr-31.md (hunk 5 above)
Comment (srallen) [reply docs/arch/adr-31.md:57]: Ok. I think in all likelihood based on our file size limitations that we won't have subjects with minutes.

**Pull request merged**